processing/src/main/java/org/apache/druid/query/ResourceLimitExceededException.java

```diff
@@ -19,18 +19,33 @@
 
 package org.apache.druid.query;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import org.apache.druid.java.util.common.StringUtils;
 
 /**
  * Exception indicating that an operation failed because it exceeded some configured resource limit.
  *
- * This is used as a marker exception by {@link QueryInterruptedException} to report the "Resource limit exceeded"
- * error code.
+ * This is a {@link BadQueryException} because it will likely a user's misbehavior when this exception is thrown.
+ * Druid cluster operators set some set of resource limitations for some good reason. When a user query requires
+ * too many resources, it will likely mean that the user query should be modified to not use excessive resources.
```

Comment:
Based on the description, shouldn't [this exception](https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/client/DirectDruidClient.java#L447) also be categorized as ResourceLimitExceeded ?

Reply:
Good catch :slightly_smiling_face: It should be, but I wasn't 100% sure whether it could cause any side effects if I did it. I left a comment about it [here](https://github.com/apache/druid/pull/10746/files#diff-28a786f3719500e89592ff9617a926736337b723e4245c410601669db0f29ee5R171). I will also make an issue to track it.

Reply:
If we change this, the exception type and code changes from RE to ResourceLimitExceeded for queries exceeding maxScatterGatherBytes. What are the side effects that could happen? Sorry if I'm missing something :)

Reply:
@a2l007 oh, you are correct about the exception type in `checkTotalBytesLimit()`. My point was that this exception doesn't seem to be propagated to `JsonParserIterator` anyway because [`NettyHttpClient` doesn't do it](https://github.com/apache/druid/blob/master/core/src/main/java/org/apache/druid/java/util/http/client/NettyHttpClient.java#L263). I think `NettyHttpClient` should call `retVal.setException()` instead of `retVal.set(null)`, but I haven't studied the code enough around it to figure out why we are doing this. Anyway, it doesn't look harm to change the exception type in this PR, so I changed :slightly_smiling_face: 

@jon-wei :+1: Updated the javadoc.

Reply:
@jihoonson Makes sense. Thanks for clarifying!